### PR TITLE
Remove legacy CLI, code, and scaffold templates

### PR DESCRIPTION
**Pull request by joelanford**
**Description of the change:**
Removes:
- [x] `operator-sdk add api` (for Go projects)
- [x] `operator-sdk add controller` 
- [x] `operator-sdk migrate`
- [x] `operator-sdk print-deps`
- [x] `operator-sdk generate k8s`
- [x] `operator-sdk generate crds`
- [x] The legacy scorecard implementation
- [x] The scorecard proxy image build.
- [x] The legacy e2e tests
- [x] Legacy scaffold templates

**Motivation for the change:**
Remove legacy code in preparation for 1.0 release

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -234,38 +234,6 @@ jobs:
...
-        - make image-push-scorecard-proxy
-
-    # Build and deploy s390x scorecard-proxy docker image
-    - <<: *deploy
-      name: Docker image for scorecard-proxy (s390x)
-      arch: s390x
-      script:
-        - make image-build-scorecard-proxy
-        - make image-push-scorecard-proxy
-
```
Comment (jmrodri) [inline .travis.yml:268]: :heart: That's one less set of images to fail during a build :)

File: go.mod [hunk 2]
```diff
@@ -13,13 +13,11 @@ require (
 	github.com/markbates/inflect v1.0.4
 	github.com/mattn/go-isatty v0.0.12
 	github.com/mitchellh/go-homedir v1.1.0
-	github.com/mitchellh/mapstructure v1.1.2
 	github.com/onsi/ginkgo v1.12.0
 	github.com/onsi/gomega v1.9.0
 	github.com/operator-framework/api v0.3.8
 	github.com/operator-framework/operator-registry v1.12.6-0.20200611222234-275301b779f8
 	github.com/pborman/uuid v1.2.0
-	github.com/pkg/errors v0.9.1
```
Comment (jmrodri) [inline go.mod:22]: Surprising this must have been used only in a few places.

File: Makefile [hunk 3]
```diff
@@ -279,15 +279,12 @@ test-subcommand-olm-install:
...
 
-test-e2e: test-e2e-go test-e2e-go-new test-e2e-ansible test-e2e-ansible-molecule test-e2e-helm ## Run the e2e tests
+test-e2e: test-e2e-go test-e2e-ansible test-e2e-ansible-molecule test-e2e-helm ## Run the e2e tests
 
 test-e2e-go:
-	./hack/tests/e2e-go.sh $(ARGS)
-
-test-e2e-go-new:
-	K8S_VERSION=$(K8S_VERSION) ./hack/tests/e2e-go-new.sh
+	K8S_VERSION=$(K8S_VERSION) ./hack/tests/e2e-go.sh
```
Comment (jmrodri) [inline Makefile]: +1 made sure the targets were repurposed.

File: hack/generate/test-framework/gen-test-framework.sh [hunk 4]
```diff
@@ -18,5 +18,4 @@ go build ./...
 go mod tidy
 
 # Run gen commands
-../../build/operator-sdk generate k8s
```
Comment (jmrodri) [inline hack/generate/test-framework/gen-test-framework.sh:23]: good idea

File: release.sh [hunk 5]
```diff
@@ -38,25 +38,6 @@ if [[ "$VER" != "$CURR_VER" ]]; then
...
-if [[ "$VER" != "$CURR_VER_ANS_GOMOD" ]]; then
-	echo "go.mod 'replace' entry version is not set correctly in $ANS_GOMOD"
-	exit 1
-fi
-CURR_VER_HELM_GOMOD="$(sed -E -n -r 's|.*operator-sdk ([^ \t\n]+).*|\1|p' "$HELM_GOMOD" | tail -1 | tr -d ' \t\n')"
-if [[ "$VER" != "$CURR_VER_HELM_GOMOD" ]]; then
-	echo "go.mod 'replace' entry version is not set correctly in $HELM_GOMOD"
-	exit 1
-fi
-
```
Comment (jmrodri) [inline release.sh:59]: Nicely done

File: .travis.yml [hunk 6]
```diff
@@ -51,9 +51,9 @@ x_base_steps:
       - make setup-k8s
       - export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
     after_success:
-      - echo "Build succeeded, operator was generated, memcached operator is running on $CLUSTER, and unit/integration tests pass"
+      - echo "Tests passed"
```
Comment (camilamacedo86) [inline .travis.yml:54]: great 👍

File: .travis.yml (hunk 1 above)
Comment (camilamacedo86) [reply .travis.yml:268]: but do you not need these images?

File: go.mod [hunk 7]
```diff
@@ -13,13 +13,11 @@ require (
 	github.com/markbates/inflect v1.0.4
 	github.com/mattn/go-isatty v0.0.12
 	github.com/mitchellh/go-homedir v1.1.0
-	github.com/mitchellh/mapstructure v1.1.2
```
Comment (camilamacedo86) [inline go.mod:16]: great 👍

File: internal/scaffold/constants.go [hunk 8]
```diff
@@ -23,15 +23,17 @@ const (
...
+	ApisDir       = PkgDir + filePathSep + "apis"
+	ControllerDir = PkgDir + filePathSep + "controller"
+	BuildDir      = "build"
+	BuildBinDir   = BuildDir + filePathSep + "_output" + filePathSep + "bin"
+	DeployDir     = "deploy"
+	CRDsDir       = DeployDir + filePathSep + "crds"
+
+	// files
+	DockerfileFile   = "Dockerfile"
+	OperatorYamlFile = "operator.yaml"
```
Comment (camilamacedo86) [inline internal/scaffold/constants.go:38]: Why we need to keep the `OperatorYamlFile ?

File: internal/scaffold/constants.go [hunk 9]
```diff
@@ -23,15 +23,17 @@ const (
...
-	CRDsDir        = DeployDir + filePathSep + "crds"
+	CmdDir        = "cmd"
+	ManagerDir    = CmdDir + filePathSep + "manager"
+	PkgDir        = "pkg"
+	ApisDir       = PkgDir + filePathSep + "apis"
+	ControllerDir = PkgDir + filePathSep + "controller"
+	BuildDir      = "build"
+	BuildBinDir   = BuildDir + filePathSep + "_output" + filePathSep + "bin"
+	DeployDir     = "deploy"
+	CRDsDir       = DeployDir + filePathSep + "crds"
```
Comment (camilamacedo86) [inline internal/scaffold/constants.go:34]: Are not they valid just for the legacy layout?

File: internal/scaffold/constants.go (hunk 8 above)
Comment (camilamacedo86) [inline internal/scaffold/constants.go:38]: we will be able to remove the vars related to the legacy layout. However, it is not in the scope of this pr.

File: .travis.yml (hunk 1 above)
Comment (jmccormick2001) [reply .travis.yml:268]: the scorecard-proxy is not used in the new scorecard so its fine to remove.

**Comment by drewwells** [comment]
For the record, I just started using this project 2 months ago and the CLI has completely changed and there was no upgrade path offered in the cli itself. This is a major letdown as a user. We have a scaffolding tool for our internal projects and it suffers the same issues with upgrades. Scaffolding may not be the best way to offer general purpose toolkits for people to build projects on.

**Comment by estroz** [comment]
@drewwells I’m sorry to hear that we’ve let you down. We do not enjoy breaking things for users, and didn’t make the decision to do so lightly.

I and many other SDK contributors felt these changes necessary to improve maintainability of scaffolded projects and be better tenants of the k8s community (ex. align with kubebuilder’s scaffolds, use kustomize extensively). Going forward we will not make any breaking changes to `operator-sdk`’s CLI or project scaffolds in any minor release, and do not plan on making drastic CLI changes in the next major release (which is still far off to begin with).

Unfortunately it isn’t feasible to write a full migration tool due to the complexity of these changes. However we do have a [migration guide](https://sdk.operatorframework.io/docs/building-operators/golang/project_migration_guide/) that covers migrating both CLI and projects to their v1 states. Feel free to open issues for specific questions about the new CLI/project layout too!

**Comment by drewwells** [comment]
Thanks for the acknowledgement. We've started the migration and it has gone well. I'll try to find time to commit to the docs to help with some gotchas that I ran into. Thanks for the great work, it's been a great toolkit for us!

**Pull request merged**